From d58d4077bda033d11496c116eda8ea181a8e5ac0 Mon Sep 17 00:00:00 2001
From: Michal Rostecki <mrostecki@opensuse.org>
Date: Mon, 11 Nov 2019 16:01:05 +0100
Subject: [PATCH 2/2] Stackdriver stats: Return false in matcher when type is
 unknown

Before this change, when type was unknown, MatchAndExplain method could
return random data. This change adds default return of false value when
type does not match any handled bases.
---
 opencensus/exporters/stats/stackdriver/internal/time_series_matcher.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/opencensus/exporters/stats/stackdriver/internal/time_series_matcher.cc b/opencensus/exporters/stats/stackdriver/internal/time_series_matcher.cc
index 47618b8..28254bb 100644
--- a/opencensus/exporters/stats/stackdriver/internal/time_series_matcher.cc
+++ b/opencensus/exporters/stats/stackdriver/internal/time_series_matcher.cc
@@ -124,6 +124,8 @@ bool TimeSeriesMatcher::MatchAndExplain(
       return true;
     }
   }
+  *listener << "has unknown type";
+  return false;
 }
 
 void TimeSeriesMatcher::DescribeTo(::std::ostream* os) const {
-- 
2.16.4

